src/item_contents.cpp

```diff
@@ -786,6 +787,40 @@ int item_contents::remaining_capacity_for_liquid( const item &liquid ) const
     return charges_of_liquid;
 }
 
+float item_contents::relative_encumbrance() const
+{
+    units::volume nonrigid_capacity;
+    units::volume nonrigid_volume;
+    for( const item_pocket &pocket : contents ) {
+        if( !pocket.is_type( item_pocket::pocket_type::CONTAINER ) ) {
+            continue;
+        }
+        if( pocket.rigid() ) {
+            continue;
+        }
+        nonrigid_volume += pocket.contains_volume();
+        nonrigid_capacity += pocket.volume_capacity();
+    }
+    assert( nonrigid_volume <= nonrigid_capacity );
+    if( nonrigid_capacity == 0_ml ) {
+        return 0;
+    }
+    return nonrigid_volume * 1.0 / nonrigid_capacity;
+}
+
+bool item_contents::is_rigid() const
```

Comment:
i think maybe this should be renamed to any_pocket_is_rigid() just to make sure it's used properly by any additional contributors?